### PR TITLE
split ScenarioObjective::Objective into 2 parts, so part of it can be reused in S5::Options

### PR DESCRIPTION
**Pull request by LeftofZen**
Suggested here:
_Originally posted by @duncanspumpkin in https://github.com/OpenLoco/OpenLoco/pull/1487#discussion_r861452305_

### CODE REVIEW

File: src/OpenLoco/ScenarioObjective.h [hunk 1]
```diff
@@ -8,19 +8,27 @@ namespace OpenLoco::Scenario
...
+        uint8_t deliveredCargoType;    // 0x000423 (0x0052623B)
+        uint32_t deliveredCargoAmount; // 0x000424 (0x0052623C)
+        uint8_t timeLimitYears;        // 0x000428 (0x00526240)
+    };
+#pragma pack(pop)
+    static_assert(sizeof(Objective) == 0x11);
+    Objective& getObjective();
+
+#pragma pack(push, 1)
+    struct Objective2
```
Comment (AaronVanGeffen) [inline src/OpenLoco/ScenarioObjective.h:25]: ObjectiveProgress?

Comment (LeftofZen) [reply src/OpenLoco/ScenarioObjective.h:25]: sounds good, I didn't know what to call it!

Comment (LeftofZen) [reply src/OpenLoco/ScenarioObjective.h]: done

File: src/OpenLoco/EditorController.cpp [hunk 2]
```diff
@@ -87,17 +87,17 @@ namespace OpenLoco::EditorController
...
-        options.objectiveCompanyValue = Scenario::getObjective().companyValue;
-        options.objectiveMonthlyVehicleProfit = Scenario::getObjective().monthlyVehicleProfit;
-        options.objectivePerformanceIndex = Scenario::getObjective().performanceIndex;
-        options.objectiveDeliveredCargoType = Scenario::getObjective().deliveredCargoType;
-        options.objectiveDeliveredCargoAmount = Scenario::getObjective().deliveredCargoAmount;
-        options.objectiveTimeLimitYears = Scenario::getObjective().timeLimitYears;
-
-        options.objectiveDeliveredCargo = ObjectManager::getHeader(LoadedObjectHandle{ ObjectType::cargo, options.objectiveDeliveredCargoType });
-        options.currency = ObjectManager::getHeader(LoadedObjectHandle{ ObjectType::currency, options.objectiveDeliveredCargoType });
+        options.objective.type = Scenario::getObjective().type;
```
Comment (duncanspumpkin) [inline src/OpenLoco/EditorController.cpp:90]: cant this become `options.objective = Scenario.getObjective();`

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**